packages/flutter_tools/lib/src/dart/dependencies.dart

```diff
@@ -27,10 +43,26 @@ class DartDependencySetBuilder {
       final Uri currentUri = toProcess.removeLast();
       final analyzer.CompilationUnit unit = _parse(currentUri.toFilePath());
       for (analyzer.Directive directive in unit.directives) {
-        if (!(directive is analyzer.UriBasedDirective))
+        String uriAsString;
+        if (directive is analyzer.UriBasedDirective) {
+          if (directive is analyzer.NamespaceDirective) {
+            final analyzer.NamespaceDirective namespaceDirective = directive;
+            // If the directive is a conditional import directive, we should
+            // select the imported uri based on the condition.
+            for (analyzer.Configuration configuration in namespaceDirective.configurations) {
+              if (_configurationConstants.contains(_dottedNameToString(configuration.name))) {
+                uriAsString = configuration.uri.stringValue;
+                break;
+              }
+            }
+          }
+          if (uriAsString == null) {
+            final analyzer.UriBasedDirective uriBasedDirective = directive;
+            uriAsString = uriBasedDirective.uri.stringValue;
+          }
+        } else {
```

Comment:
To make this slightly more readable and save one level of indention, could you turn this if arround:
```
if (foo)
   continue
do other stuff
```